libraries/Microsoft.Bot.Builder.Core.Extensions/TimeoutWarningMiddleware.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
```

Comment:
Consider generalizing to not just warn on an error. 

It would be nice to get basic data such as "how long?" things took to run. If the ctor took a lamnda that is called on "completed", that method could do whatever the user needed as long as the "latency" is passed in at the parameter.

Reply:
Also, need some tests. :)

Reply:
@cleemullins this was based on #36 opened by @Stevenic.  I think the idea wasn't to necessarily let the user do anything on a timeout, but simply to trace a warning for them to see in the logs to alert them that a timeout was potentially going to occur after a given period, which can obviously cause problems with channels like Cortana.

Interested to open the discussion though to see if we think it should be something else?